FF1Lib/EnemyFormations.cs

```diff
@@ -81,8 +63,13 @@ public void ShuffleUnrunnable(MT19337 rng)
 			// Generate a shuffled list of ids for encounters
 			// We include - all normal formation A-Sides except encounter 00 (imps), all normal formation B-Sides, and the four B-sides at the end
 			List<int> ids = Enumerable.Range(1, NormalFormationCount - 1).Concat(Enumerable.Range(FormationCount, NormalFormationCount)).Concat(Enumerable.Range(FormationCount + ChaosFormationIndex + 1, 4)).ToList();
-			ids.Shuffle(rng);
-			ids = ids.Take(unrunnableAcount + unrunnableBcount).ToList(); // combine the number of unrunnables between both sides
+			ids.Shuffle(rng);
+
+			// Take unrunnable percentage from flags and convert into number of encounters that are unrunnable
+			// The +1 to NormalFormationCount is needed to account for the extra B-side formation spots
+			int unrunnableTotal = (int)Math.Round((double)(unrunnablePercent * (NormalFormationCount + 1) / 100 * 2));

```

Comment:
I'm going to guess you attempted to do accurate math here. However the cast to double and the rounding is done after the evaluation othe the formular. Thus everything is done with integer math.
The Multiplication and addition isn't a problem anyway. Just change 100 to 100.0 to force it to a double precision division. You can then remove the explicit cast to double.